### ISSUE TITLE
Qwen2.5-VL-7B-Instruct output unexpected results "addCriterion"

### ISSUE THREAD

**Issue by SunnyLi2015** [QwenLM/Qwen2.5-VL#759]
Hi,

I am trying to run 7B VLM model with the example shows https://huggingface.co/Qwen/Qwen2.5-VL-7B-Instruct, found several issues

- when I use **a detailed prompt**, the model sometimes output
"addCriterion
对不起，我无法理解您提供的信息或
 addCriterion
好的，请提供更多信息以便我更好地帮助您。"
It happens in 1/8  cases. 

- when I use a **naïve prompt** as "请生成对于图片的描述", the results are better, but always get chattering with "addCriterion"  

The results are ok with online 72B model in both cases.

Do you have any idea? Thanks~

**Comment by Bill2462** [comment]
I encountered the same problem with 8B version.

I think this might be due to batching. If I tried example with batch size of 1 everything works fine. If I however try a batched example with multiple inputs, it starts to output addCriterion and the quality of the output also decreases significantly. It for example does not follow the prompt to output only in English and begins to randomly output in Chinese.

**Comment by SunnyLi2015** [comment]
Thanks @Bill2462 for your input. In my case, I don't get any error/warning, and I am using batch size = 1.

**Comment by xiankgx** [comment]
Same problem here!

**Comment by Elenore1997** [comment]
Same situation here. Seems it occurs when inferencing with bs > 1.

**Comment by HiXxxSss** [comment]
When I using model to classification, have same problem!

**Comment by yantijin** [comment]
same problem here!

**Comment by skyfreetree** [comment]
Same problem here!

**Comment by eav-solution** [comment]
Same problem if I used vllm with tool calling.
 "content": "<tool_call>\n addCriterion\n<tool_call>\n\n addCriterion\n\n\n addCriterion\n\n addCriterion\n\n\n addCriterion\n\n\n\n\n\n\n\n\n\n<tool_call>\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n<tool_call>\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n",

**Comment by junyeon-nero** [comment]
Same problem here!

**Comment by 2013358072** [comment]
Same problem! 
 prompt= f"Below is the image of one page of a document\n"
    f"Just return the plain text representation of this document as if you were reading it naturally.\n"
    f"Do not hallucinate.\n"
result: 
 addCriterion

 addCriterion

 addCriterion

 addCriterion

 addCriterion

 addCriterion

 addCriterion

**Comment by WGowi** [comment]
Same problem here!  同样有这个问题！

**Comment by junkerman2004** [comment]
大概率是因为generate代码的原因：processor的问题
可以参照下面这段代码：用自web—demo
prompt,image_paths=self.get_query(part_questions)
            
            text= processor.apply_chat_template(prompt, tokenize=False, add_generation_prompt=True)
            images,videos=process_vision_info(prompt)
            #images = []
            #for path in image_paths:
                #print(path)
                #try:
                    #images.append(Image.open(path))
                #except PIL.UnidentifiedImageError:
                    #print(f"Skipped invalid image: {path}")
            device = next(self.model.parameters()).device  # 获取模型所在的设备
            inputs = self.processor(text=[text], images=images, padding=True, return_tensors="pt").to(device)
            tokenizer = processor.tokenizer
            streamer = TextIteratorStreamer(tokenizer, timeout=20.0, skip_prompt=True, skip_special_tokens=True)

            gen_kwargs = {'max_new_tokens': 512, 'streamer': streamer, **inputs}
            thread = Thread(target=model.generate, kwargs=gen_kwargs)
            thread.start()

            output_text = ''
            for new_text in streamer:
                output_text += new_text
可以解决困扰您的生成问题

**Comment by junyeon-nero** [comment]
@junkerman2004 thank you for your comments! I solved this issue with a single line of code : `device = next(self.model.parameters()).device`

**Comment by eav-solution** [comment]
> [@junkerman2004](https://github.com/junkerman2004) thank you for your comments! I solved this issue with a single line of code : `device = next(self.model.parameters()).device`

I dont know how to fix it in vllm, can you suggest me ?

**Comment by junyeon-nero** [comment]
@eav-solution I didn't use vllm but I can provide HF inference code.

```
import torch
from transformers import Qwen2_5_VLForConditionalGeneration, AutoProcessor
from qwen_vl_utils import process_vision_info

model_name = "Qwen/Qwen2.5-VL-7B-Instruct"

processor = AutoProcessor.from_pretrained(model_name, trust_remote_code=True)
model = Qwen2_5_VLForConditionalGeneration.from_pretrained(model_name, torch_dtype="auto", device_map="auto").eval()

from models.PROMPT import prompt_v1, prompt_v2
prompt = prompt_v2

def convert(path):
    messages = [{"role": "user", "content": [{"type": "image", "image": f"file://{path}"}, {"type": "text", "text": prompt}]}]
    text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
    image_inputs, video_inputs = process_vision_info(messages)
    
    device = next(model.parameters()).device 
    inputs = processor(text=[text], images=image_inputs, videos=video_inputs, padding=True, return_tensors="pt").to(device)
    with torch.inference_mode():
        generated_ids = model.generate(**inputs, max_new_tokens=2048)
        generated_ids_trimmed = [out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)]
        output_text = processor.decode(generated_ids_trimmed[0], skip_special_tokens=True, clean_up_tokenization_spaces=False)
        return output_text
    return ""
```

**Comment by Elenore1997** [comment]
> [@eav-solution](https://github.com/eav-solution) I didn't use vllm but I can provide HF inference code.
> 
> ```
> import torch
> from transformers import Qwen2_5_VLForConditionalGeneration, AutoProcessor
> from qwen_vl_utils import process_vision_info
> 
> model_name = "Qwen/Qwen2.5-VL-7B-Instruct"
> 
> processor = AutoProcessor.from_pretrained(model_name, trust_remote_code=True)
> model = Qwen2_5_VLForConditionalGeneration.from_pretrained(model_name, torch_dtype="auto", device_map="auto").eval()
> 
> from models.PROMPT import prompt_v1, prompt_v2
> prompt = prompt_v2
> 
> def convert(path):
>     messages = [{"role": "user", "content": [{"type": "image", "image": f"file://{path}"}, {"type": "text", "text": prompt}]}]
>     text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
>     image_inputs, video_inputs = process_vision_info(messages)
>     
>     device = next(model.parameters()).device 
>     inputs = processor(text=[text], images=image_inputs, videos=video_inputs, padding=True, return_tensors="pt").to(device)
>     with torch.inference_mode():
>         generated_ids = model.generate(**inputs, max_new_tokens=2048)
>         generated_ids_trimmed = [out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)]
>         output_text = processor.decode(generated_ids_trimmed[0], skip_special_tokens=True, clean_up_tokenization_spaces=False)
>         return output_text
>     return ""
> ```

have u tried batch_size > 2? i have tried ur method but still got empty result in a batch data.

**Comment by junyeon-nero** [comment]
> > [@eav-solution](https://github.com/eav-solution) I didn't use vllm but I can provide HF inference code.
> > ```
> > import torch
> > from transformers import Qwen2_5_VLForConditionalGeneration, AutoProcessor
> > from qwen_vl_utils import process_vision_info
> > 
> > model_name = "Qwen/Qwen2.5-VL-7B-Instruct"
> > 
> > processor = AutoProcessor.from_pretrained(model_name, trust_remote_code=True)
> > model = Qwen2_5_VLForConditionalGeneration.from_pretrained(model_name, torch_dtype="auto", device_map="auto").eval()
> > 
> > from models.PROMPT import prompt_v1, prompt_v2
> > prompt = prompt_v2
> > 
> > def convert(path):
> >     messages = [{"role": "user", "content": [{"type": "image", "image": f"file://{path}"}, {"type": "text", "text": prompt}]}]
> >     text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
> >     image_inputs, video_inputs = process_vision_info(messages)
> >     
> >     device = next(model.parameters()).device 
> >     inputs = processor(text=[text], images=image_inputs, videos=video_inputs, padding=True, return_tensors="pt").to(device)
> >     with torch.inference_mode():
> >         generated_ids = model.generate(**inputs, max_new_tokens=2048)
> >         generated_ids_trimmed = [out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)]
> >         output_text = processor.decode(generated_ids_trimmed[0], skip_special_tokens=True, clean_up_tokenization_spaces=False)
> >         return output_text
> >     return ""
> > ```
> 
> have u tried batch_size > 2? i have tried ur method but still got empty result in a batch data.

no I just use batch size = 1

**Comment by SunnyLi2015** [comment]
> [@junkerman2004](https://github.com/junkerman2004) thank you for your comments! I solved this issue with a single line of code : `device = next(self.model.parameters()).device`

I can still reproduce the issue

**Comment by hlchen23** [comment]
set padding_side='left' when batch size > 1, or only the longest prompt will not be interfered.

**Comment by tomato654** [comment]
none of solution is working for me.
The model even outputs following answer using lmms-eval mmmu_val:
['To determine the correct order in which the key values could have been inserted into the hash table using linear\n addCriterion\nlinear probing, we need\n addCriterion\nneed to ensure that each insertion follows the rule of linear probing, meaning if\n addCriterion\nif a slot is occupied by another value, the next available slot is chosen sequentially until an empty slot is found\n addCriterion\nfound.\n\nGiven\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n']

**Comment by hlchen23** [comment]
您的来件已收到，谢谢！（自动回复）Chen

**Comment by hiskiv** [comment]
I also encountered the problem and found out that when I placed my image tokens (<image>) in the middle of a long prompt the output also yielded many 'addCriterion'. After I moved all the image tokens to the beginning of the prompt it just worked fine.

**Comment by hlchen23** [comment]
您的来件已收到，谢谢！（自动回复）Chen

**Comment by William030422** [comment]
Same problem here. It works fine for me if I do not specify output_attentions=True. But if I specify this, it is going to output tones of addCriterion.

**Comment by YANGTUOMAO** [comment]
this is an unbelievable bug, i feel shoke that such bug still exist! 
这傻逼bug还没修复吗？

**Comment by William030422** [comment]
lol it seems so 🙌

**Comment by MoyusiteruIori** [comment]
I'm running the model on a single GPU with batch size 1, yet I also encounter this problem.

**Comment by DMRookie** [comment]
> Thanks [@Bill2462](https://github.com/Bill2462) for your input. In my case, I don't get any error/warning, and I am using batch size = 1.

我这边是padding的问题，通过设置 processor.tokenizer.padding_side = "left" 可以修复。

测试过batch内的图片大小差不多时，都可以正常输出。差别比较大，有的就会输出addCriterion。
只有单张图片时，可能是minpixel设置过大？

**Comment by YANGTUOMAO** [comment]
我已经改回qwen2-vl了！！！

**Comment by felmoreno1726** [comment]
I was hitting this issue as well. It was resolved for me after setting the Chat template to Qwen2-VL. This was in SGLang but it might be relevant to other generation frameworks.

**Comment by zzj-hem** [comment]
> Thanks [@Bill2462](https://github.com/Bill2462) for your input. In my case, I don't get any error/warning, and I am using batch size = 1.

inputs = processor(  # 将文本、图像和视频输入组合成一个输入字典，并将其转换为PyTorch张量。
    text=texts,
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
    padding_side='left'
)  应该就是这个原因，加上这个参数padding_side='left',就能解决了

**Comment by xxhhbbbb** [comment]
Same problem! 
I encountered this issue while trying to implement attention visualization for Qwen2.5-VL. 
Finally, I found that changing the attn_implementation from "eager" to 'flash_attention_2' appears to resolve the problem and the model starts responding normally.

**Comment by hlchen23** [comment]
您的来件已收到，谢谢！（自动回复）Chen

**Comment by danielkorth** [comment]
Had the same issue (addCriterion). When I enabled `flash_attention_2` I got chinese signs. Now when I additionally set the padding_side to left, it finally seems to work.

**Comment by buffliu** [comment]
Have anybody solved this problem?

**Comment by thiswillbeyourgithub** [comment]
Have you tried not enabling flash attention?

**Comment by buffliu** [comment]
HI @thiswillbeyourgithub, Thank you! 

I probably didn't enable flash attention. The code is as follows.
 model = Qwen2_5_VLForConditionalGeneration.from_pretrained(model_path,
                                                            load_in_low_bit="sym_int8",
                                                            optimize_model=True,
                                                            trust_remote_code=True,
                                                            modules_to_not_convert=["vision"],
                                                            use_cache=True,
                                                            model_hub=model_hub)

when i switch Qwen-VL-3B-INTRSUTC, it works well.

**Comment by thiswillbeyourgithub** [comment]
Enabling flash attention is done via the environment variables.

**Comment by buffliu** [comment]
@thiswillbeyourgithub,  Could you tell me how to set the variable? thanks a lot!

**Comment by thiswillbeyourgithub** [comment]
na too busy use LLMs

**Comment by huaiyizhao** [comment]
pdding_size='left' solves the problem for me. If using attn_implementation="flash_attention_2" when loading the model, it actually raises an error if not setting pdding_size='left' explicitly. Don't know why the default padding side is right in the processor.

**Comment by lucasmgomez** [comment]
> Same problem! I encountered this issue while trying to implement attention visualization for Qwen2.5-VL. Finally, I found that changing the attn_implementation from "eager" to 'flash_attention_2' appears to resolve the problem and the model starts responding normally.

I am trying to do the same, but hugginface seems to make flash_attention_2 incompatible with the .generate setting "output_attentions". Did you find a simple way to get around this?

**Comment by Louis-61** [comment]
`processor = AutoProcessor.from_pretrained("Qwen/Qwen2.5-VL-7B-Instruct", padding_side='left')`
this can solve the problem because i see a warning says "A decoder-only architecture is being used, but right-padding was detected! For correct generation results, please set 'padding_side='left'' when initializing the tokenizer."

**Comment by tommy-fcy** [comment]
`padding_size='left'` might not fix this problem under the newest version of transformers like `transformers==4.55`, reinstall `transformers==4.52.3` can solve this problem

**Comment by bghira** [comment]
yep it's broken in newest transformers

**Comment by LeviiveLl** [comment]
> Same problem! I encountered this issue while trying to implement attention visualization for Qwen2.5-VL. Finally, I found that changing the attn_implementation from "eager" to 'flash_attention_2' appears to resolve the problem and the model starts responding normally.

i use attn_implementation="flash_attention_2" all the time, but the problem still exsists

**Comment by shepard2k** [comment]
> `padding_size='left'` might not fix this problem under the newest version of transformers like `transformers==4.55`, reinstall `transformers==4.52.3` can solve this problem

Thank you so much. That fixed it for me.

**Comment by hlchen23** [comment]
您的来件已收到，谢谢！（自动回复）Chen

**Comment by bghira** [comment]
the flash-attention is broken by the way, you're stuck with xformers

**Comment by hlchen23** [comment]
您的来件已收到，谢谢！（自动回复）Chen

**Comment by Clementine24** [comment]
I still meet the same problem, and I use vLLM.

**Comment by hteeyeoh** [comment]
I'm still seeing issue after installing transformers==4.52.3. 
Any suggestion here?

**Comment by AngelAlita** [comment]
Removed `output_scores=True` and `output_attentions=True` inside `model.generate(...)` fixes it for me.
